src/praisonai/praisonai/auto.py

```
@@ -1137,45 +1130,7 @@ def __init__(self, topic: str = "Research and write about AI",
-            return "evaluator-optimizer"
-        elif any(kw in task_lower for kw in orchestrator_keywords):
-            return "orchestrator-workers"
-        elif any(kw in task_lower for kw in routing_keywords):
-            return "routing"
-        elif any(kw in task_lower for kw in parallel_keywords):
-            return "parallel"
-        else:
-            return "sequential"
-    
-    def recommend_pattern_llm(self, topic: str = None):  # Returns PatternRecommendation at runtime
+    def recommend_pattern_llm(self, topic: str = None) -> PatternRecommendation:
```

Ruff (0.15.17)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)